.github/workflows/stacks-blockchain.yml

```diff
@@ -0,0 +1,175 @@
+name: stacks-blockchain
+
+# Only run when:
+#   - tags starting with "v" get pushed
+#   - PRs are opened against the master branch
+#   - the workflow is started from the UI (an optional tag can be passed in via parameter)
+#     - If the optional tag parameter is passed in, a new tag will be generated based off the selected branch
+on:
+  push:
+    tags:
+      - 'v*'
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'The tag to create (optional)'
+        required: false
+
+jobs:
+  # Notify Slack channel of workflow start
+  notify-start:
+    runs-on: ubuntu-latest
+    outputs:
+      slack_message_id: ${{ steps.slack.outputs.message_id }}
+    steps:
+      - name: Notify slack start
+        id: slack
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+        uses: voxmedia/github-action-slack-notify-build@v1.1.2
+        with:
+          channel: devops-notify
+          status: STARTING
+          color: warning
+
+  # Run tests
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run bitcoin tests
+        env:
+          DOCKER_BUILDKIT: 1
+        run: docker build -f ./.github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests .
+
+  # Create distributions
+  dist:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [ windows-x64, macos-x64, linux-x64, linux-musl-x64, linux-armv7, linux-arm64 ]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build distributable
+        env:
+          DIST_TARGET_FILTER: ${{ matrix.platform }}
+        run: build-scripts/build-dist.sh
+
+      - name: Compress artifact
+        run: zip --junk-paths ${{ matrix.platform }} ./dist/${{ matrix.platform }}/*
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.platform }}
+          path: ${{ matrix.platform }}.zip
+
+  # Build docker image, tag it with the git tag and `latest` if running on master branch, and publish under the following conditions
+  # Will publish if:
+  #   - a tag was passed into this workflow
+  #   - a tag was pushed up
+  #   - this workflow was invoked against a non-master branch (a Docker image tag with the name of the branch will be published)
+  build-publish:
```

Comment:
Can this be updated to build both `Dockerfile` and `Dockerfile.stretch`?

Reply:
@kantai Should it also publish the `Dockerfile.stretch` image under a different tag? Or should it just build the stretch image as a test and never publish?

Reply:
Yeah -- it should publish it as `stacks-blockchain:vX.Y.Z-stretch`

Reply:
This workflow will now build and publish (when appropriate) the stretch Dockerfile